### PR TITLE
add ipld to amt::array in init actor construct method

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey-N-Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
amt::Array requires Ipld on construction, the PR adds it to Init actor construct method.

### CODE REVIEW

File: test/core/vm/actor/builtin/v0/init/init_actor_test.cpp [hunk 1]
```diff
@@ -15,48 +15,85 @@
...
+  }
+
+  /**
+   * @given caller is system actor
+   * @when construct is called
+   * @then init actor is constructed
+   */
+  TEST_F(InitActorTest, ConstructSuccess) {
+    caller = kSystemActorAddress;
+    std::string network_name = "test_network_name";
```
Comment (wer1st) [inline test/core/vm/actor/builtin/v0/init/init_actor_test.cpp:72]:
```suggestion
    const std::string network_name = "test_network_name";
```

File: core/vm/actor/builtin/v0/init/init_actor.cpp [hunk 2]
```diff
@@ -23,8 +23,9 @@ namespace fc::vm::actor::builtin::v0::init {
 
   ACTOR_METHOD_IMPL(Construct) {
     OUTCOME_TRY(runtime.validateImmediateCallerIs(kSystemActorAddress));
-    InitActorState state{};
-    state.network_name = params.network_name;
+    InitActorState state{.address_map = {runtime.getIpfsDatastore()},
```
Comment (wer1st) [inline core/vm/actor/builtin/v0/init/init_actor.cpp:26]: Have you checked other amt containers? (map, balanceTable etc.)

Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/v0/init/init_actor.cpp:26]: Yes, they need the same initialisation, but all of these have already been implemented in different ways.

Comment (wer1st) [reply core/vm/actor/builtin/v0/init/init_actor.cpp:26]: Ok

**Review by wer1st — APPROVED** [review]

File: core/vm/actor/builtin/v0/init/init_actor.cpp [hunk 3]
```diff
@@ -23,8 +23,9 @@ namespace fc::vm::actor::builtin::v0::init {
 
   ACTOR_METHOD_IMPL(Construct) {
     OUTCOME_TRY(runtime.validateImmediateCallerIs(kSystemActorAddress));
-    InitActorState state{};
-    state.network_name = params.network_name;
```
Comment (turuslan) [inline core/vm/actor/builtin/v0/init/init_actor.cpp:27]: `InitActorState` implements [`Ipld::Visit`](https://github.com/filecoin-project/cpp-filecoin/blob/f42d38b6b24a010a4a7dac2e1f3bf59bdfc5319d/core/vm/actor/builtin/v0/init/init_actor.hpp#L59-L65).
Use `IpldPtr{runtime}->load(state)` to initialize amt/hamt/cidt/etc fields in constructor (other methods call `getCurrentActorStateCbor` which calls `load`)

File: test/core/vm/actor/builtin/v0/init/init_actor_test.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply test/core/vm/actor/builtin/v0/init/init_actor_test.cpp]: fixed

File: core/vm/actor/builtin/v0/init/init_actor.cpp (hunk 3 above)
Comment (Alexey-N-Chernyshov) [reply core/vm/actor/builtin/v0/init/init_actor.cpp]: ok

**Review by turuslan — APPROVED** [review]

**Pull request merged**